### ISSUE TITLE
Any update about BYOL？

### ISSUE THREAD

**Issue by cxxgtxy** [open-mmlab/mmselfsup#18]
The result remains empty. I wonder whether you have obtained some good results
Thanks.

**Comment by XiaohangZhan** [comment]
With batch size 256, epoch 200, the result is not good (VOC07 SVM 55.1%). We will launch larger experiments.

**Comment by cxxgtxy** [comment]
Keep watching your process.

**Comment by XiaohangZhan** [comment]
A bug is fixed and the results will be updated in recent days.

**Comment by cxxgtxy** [comment]
> A bug is fixed and the results will be updated in recent days.
Good news!

**Comment by cxxgtxy** [comment]
> A bug is fixed and the results will be updated in recent days.
Good news!

**Comment by cxxgtxy** [comment]
Any Update?

**Comment by cxxgtxy** [comment]
Any Update?

**Comment by XiaohangZhan** [comment]
Some results were already updated, see model_zoo.md

**Comment by cxxgtxy** [comment]
good job!
I have been watching the main readme file and lost this update.

**Comment by cxxgtxy** [comment]
You use r50_bs4096_ep200.py. Does it mean that BS 256 is hard to reproduce the result in the paper?

**Comment by XiaohangZhan** [comment]
BS256 in the paper is not actually batchsize=256. They explained: "To avoid re-tuning other hyperparameters, we average gradients over N consecutive steps before updating the online network when reducing the batch size by a factor N." It is still batchsize=4096, and the only difference lies in BatchNorm.
I have also run the experiment with BS256, and the batch size is indeed 256 to be consistent with other methods. The SVM results of r50_bs256_ep200.py in OpenSelfSup is 77.95, much lower than BS4096. I have not tried to average gradients in the way as the authors do.

**Comment by cxxgtxy** [comment]
Thanks！

**Comment by bryanyzhu** [comment]
Hi @XiaohangZhan , thank you for providing the great repo. I have two questions about BYOL. 

1. The performance in the model zoo is 67.1, which is lower than 74.3 reported in the paper. Is it because of the difference in epochs (200 vs 1000)? Any other differences causing this gap? 

2. Do you have the ImageNet lincls result of `r50_bs256_ep200.py`? I'm curious to know. 

Thank you very much.

**Comment by XiaohangZhan** [comment]
> Hi @XiaohangZhan , thank you for providing the great repo. I have two questions about BYOL.
> 
> 1. The performance in the model zoo is 67.1, which is lower than 74.3 reported in the paper. Is it because of the difference in epochs (200 vs 1000)? Any other differences causing this gap?
> 2. Do you have the ImageNet lincls result of `r50_bs256_ep200.py`? I'm curious to know.
> 
> Thank you very much.

1. number of epochs is one of the reasons. For now, I cannot find other differences.
2. It is 56.4 in the last layer, much lower than bs4096. Besides, note that bs256 here is indeed batch_size=256, which is different from that in the paper. See the discussions above.

**Comment by bryanyzhu** [comment]
> > Hi @XiaohangZhan , thank you for providing the great repo. I have two questions about BYOL.
> > 
> > 1. The performance in the model zoo is 67.1, which is lower than 74.3 reported in the paper. Is it because of the difference in epochs (200 vs 1000)? Any other differences causing this gap?
> > 2. Do you have the ImageNet lincls result of `r50_bs256_ep200.py`? I'm curious to know.
> > 
> > Thank you very much.
> 
> 1. number of epochs is one of the reasons. For now, I cannot find other differences.
> 2. It is 56.4 in the last layer, much lower than bs4096. Besides, note that bs256 here is indeed batch_size=256, which is different from that in the paper. See the discussions above.


1. Yeah, I think number of epoch is important, that makes sense. 
2. Thank you for providing the number. One thing I want to discuss further is, SimCLR actually perform batch size comparison experiment in the appendix and they seem to use the indeed bs=256. From the table below, bs=256 is lower than bs=4096, about 5% (66.6 -> 61.9) for epoch 200. But here for BYOL, bs=256 is 10% lower than bs=4096. This actually contradicts the claim from BYOL that they can do better than SimCLR in small batch size setting. What do you think of this? Thank you for any comments.

![image](https://user-images.githubusercontent.com/16021460/91592051-c7439680-e912-11ea-8da7-c2f02b905587.png)

**Comment by XiaohangZhan** [comment]
In byol, batch_size=256 is still batch_size=4096, see the discussions above: https://github.com/open-mmlab/OpenSelfSup/issues/18#issuecomment-676025354

**Comment by scnuhealthy** [comment]
Have you tried average gradients in the way as the authors do? I am curious about the performance of BYOL with the batch size of 256, although 256 is a fake size and the true size is still 4096.

**Comment by XiaohangZhan** [comment]
> Have you tried average gradients in the way as the authors do? I am curious about the performance of BYOL with the batch size of 256, although 256 is a fake size and the true size is still 4096.

Not yet. You are welcome to perform this exp.
